app/src/main/java/com/arctouch/codechallenge/view/home/HomeActivity.kt

```diff
@@ -0,0 +1,84 @@
+package com.arctouch.codechallenge.view.home
+
+import android.app.DialogFragment
+import android.app.FragmentManager
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.os.Parcelable
+import android.provider.MediaStore
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Genre
+import com.arctouch.codechallenge.view.home.detail.MovieDetailFragment
+import com.arctouch.codechallenge.viewModel.HomeViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import kotlinx.android.synthetic.main.home_activity.*
+
+class HomeActivity : AppCompatActivity() {
+    private lateinit var viewModel: HomeViewModel
+
+    private lateinit var  adapter: HomeAdapter
+
+    private var recyclerState: Parcelable? = null
+
+    private var itemSelectedSubscribe: Disposable? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setTheme(R.style.AppTheme)
+        setContentView(R.layout.home_activity)
+
+        viewModel = ViewModelProviders.of(this).get(HomeViewModel::class.java)
+        viewModel.getGenres()
```

Comment:
This is not enough to guarantee that we'll have the genres when we need them. If this call takes a long time, the first page of movies on the home page will not have genres. What would you do to make sure you only show the movies when you have fetched the genres?